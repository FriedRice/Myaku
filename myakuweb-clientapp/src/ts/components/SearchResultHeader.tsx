/** @module Header component for an article search result */

import { ArticleSearchResult } from '../types';
import React from 'react';

interface SearchResultHeaderProps {
    searchResult: ArticleSearchResult;
}
type Props = SearchResultHeaderProps;


const SearchResultHeader: React.FC<Props> = function(props) {
    return (
        <h4 className='main-tile-header japanese-text' lang='ja'>
            <a href={props.searchResult.sourceUrl}>
                {props.searchResult.title}
            </a>
        </h4>
    );
};

export default SearchResultHeader;
