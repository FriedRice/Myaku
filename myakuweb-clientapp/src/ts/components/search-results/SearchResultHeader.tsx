/** @module Header component for an article search result */

import { ArticleSearchResult } from 'ts/types/types';
import React from 'react';

interface SearchResultHeaderProps {
    searchResult: ArticleSearchResult;
}
type Props = SearchResultHeaderProps;


const SearchResultHeader: React.FC<Props> = function(props) {
    return (
        <h4 className='main-tile-header' lang='ja'>
            <a href={props.searchResult.sourceUrl}>
                {props.searchResult.title}
            </a>
        </h4>
    );
};

export default SearchResultHeader;
