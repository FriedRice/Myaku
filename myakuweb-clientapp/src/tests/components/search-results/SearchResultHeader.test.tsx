/**
 * Tests for the SearchResultHeader component.
 * @module tests/components/search-results/SearchResultHeader.test
 */

import { BASELINE_SEARCH_RESULT_PAGE } from 'tests/testData';
import React from 'react';
import SearchResultHeader from
    'ts/components/search-results/SearchResultHeader';
import { shallow } from 'enzyme';


describe('<SearchResultHeader />', function() {
    it('renders correctly', function() {
        const wrapper = shallow(
            <SearchResultHeader
                searchResult={BASELINE_SEARCH_RESULT_PAGE.results[0]}
            />
        );
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({
            searchResult: BASELINE_SEARCH_RESULT_PAGE.results[1],
        });
        expect(wrapper).toMatchSnapshot();
    });
});
