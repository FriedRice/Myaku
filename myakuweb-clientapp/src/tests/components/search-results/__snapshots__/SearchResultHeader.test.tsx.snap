// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultHeader /> renders correctly 1`] = `
<h4
  className="main-tile-header"
  lang="ja"
>
  <a
    href="https://source1.com"
  >
    Article 1
  </a>
</h4>
`;

exports[`<SearchResultHeader /> renders correctly 2`] = `
<h4
  className="main-tile-header"
  lang="ja"
>
  <a
    href="https://source2.com"
  >
    Article 2
  </a>
</h4>
`;
